spacy/lang/fr/tokenizer_exceptions.py

```diff
@@ -87,64 +87,131 @@ def lower_first_letter(text):
 for pre, pre_lemma in [
     ("qu'", "que"),
     ("n'", "ne")]:
-    for orth in [pre,pre.title()]:
+    for orth in [pre, pre.title()]:
         _exc['%sest-ce' % orth] = [
             {LEMMA: pre_lemma, ORTH: orth, TAG: "ADV"},
             {LEMMA: 'être', ORTH: "est", TAG: "VERB"},
             {LEMMA: 'ce', ORTH: '-ce'}]
 
-
 _infixes_exc = []
-for elision_char in ELISION:
-    for hyphen_char in ['-', '‐']:
-        _infixes_exc += [infix.replace("'", elision_char).replace('-', hyphen_char)
-                         for infix in FR_BASE_EXCEPTIONS]
-_infixes_exc += [upper_first_letter(word) for word in _infixes_exc]
-_infixes_exc = list(set(_infixes_exc))
+orig_elision = "'"
+orig_hyphen = '-'
+
+# loop through the elison and hyphen characters, and try to substitute the ones that weren't used in the original list
+for infix in FR_BASE_EXCEPTIONS:
```

Comment:
As before, it might be worth it to try comprehensions to make things faster : (and also change FR_BASE_EXCEPTIONS from a list to a set)
```python
_infixes_exc = FR_BASE_EXCEPTIONS.copy() #FR_BASE_EXCEPTIONS needs to be a set
_infixes_exc.update([
    word.replace(orig_elision, elision_char) 
    for elision_char in [x for x in ELISION if x != orig_elision]
    for word in _infixes_exc
])
_infixes_exc.update([
    word.replace(orig_hyphen, hyphen_char) 
    for hyphen_char in [x for x in ['-', '‐'] if x != orig_hyphen
    for word in _infixes_exc
])
_infixes_exc.update([upper_first_letter(word) for word in _infixes_exc])
_infixes_exc = list(_infixes_exc)
```

Or even try the double nested loop within the comprehension ?
```python
_infixes_exc = {
   infix.replace("'", elision_char).replace('-', hyphen_char)
   for infix in FR_BASE_EXCEPTIONS 
   for elision_char in [x for x in ELISION if x != orig_elision]
   for hyphen_char in [x for x in ['-', '‐'] if x != orig_hyphen]
}
_infixes_exc.update([upper_first_letter(word) for word in _infixes_exc])
_infixes_exc = list(_infixes_exc)
```